### DISCUSSION TITLE
Is it necessary to put images and the JSON file of emojis into the same directory?

### DISCUSSION THREAD

**Discussion by njzjz** [walinejs/waline#298] [Q&A] [answered]
表情包的图片和json文件必须在同一目录下吗？

**Comment by Mister-Hope** [comment, marked as answer]
No, in this way you should use emojioption (set info in emoji config) while setting folder to an emtpy string or a common prefix like the old emojiCDN, then add your rest part of image links in the items array.

**Comment by Mister-Hope** [comment]
The default logic is for reducing the size of info.json as it's a network request.

So we ask the users who can take full control of their server, upload their images and info.json in same folder.

In this wat we can omit the common prefix and suffix of image links in info.json, makes the info.json small in size and easy to write.
